.github/workflows/push-dev-image-on-commit.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and Push `apisix:dev` to DockerHub on Commit
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - "docs/**"
+      - "**/*.md"
+  workflow_dispatch:
```

Comment:
Here is the only difference from https://github.com/apache/apisix-docker/blob/master/.github/workflows/apisix_dev_push_docker_hub.yaml .

1. Removed the `schedule` because we are already able to ensure the latest.
2. Added `workflow_dispatch` to allow manual triggering of builds if necessary.